src/lammpsio/snapshot.py

```diff
@@ -164,31 +219,114 @@ def to_hoomd_gsd(self, type_map=None):
                 for typeidx, typeid in enumerate(sorted_typeids):
                     frame.particles.typeid[self.typeid == typeid] = typeidx
             else:
-                frame.particles.types = list(type_map.values())
-                reverse_type_map = {
-                    typeid: typeidx for typeidx, typeid in enumerate(type_map.keys())
-                }
-                for i, typeid in enumerate(self.typeid):
-                    frame.particles.typeid[i] = reverse_type_map[typeid]
+                if isinstance(type_map, dict):
+                    self.label = LabelMap(map=type_map)
```

Comment:
Here we don't want to actually override the value of `label`. In general, we also probably want to deprecate setting the `type_map` value entirely, as this should all be handled through the snapshot. Something like:
```py
if type_map is not None:
    warnings.warn(...)
    type_label_map = LabelMap(type_map)
else:
    type_label_map = self.label
```
then use this in your `_set_type_id` method